crates/polars-stream/src/nodes/parquet_source/row_group_decode.rs

```diff
@@ -138,25 +138,33 @@ impl RowGroupDecoder {
         let path_index = row_group_data.path_index;
 
         let hive_series = if let Some(hp) = self.hive_partitions.as_deref() {
-            let mut v = hp[path_index].materialize_partition_columns();
-            for s in v.iter_mut() {
-                *s = s.new_from_index(0, row_group_data.file_max_row_group_height);
-            }
-            v
+            let v = hp[path_index].materialize_partition_columns();
+            v.into_iter()
+                .map(|s| {
+                    s.into_column()
+                        .new_from_index(0, row_group_data.file_max_row_group_height)
+                })
+                .collect()
         } else {
             vec![]
         };
 
+        // @scalar-opt
         let file_path_series = self.include_file_paths.clone().map(|file_path_col| {
-            StringChunked::full(
```

Comment:
LGTM from a quick look - IIUC this does mean that (if) we materialize these they won't be shared anymore, but I don't particularly see that as a showstopper